### PR TITLE
Fix alt-tab issue with mouse move causing camera reset

### PR DESCRIPTION
**Pull request by hultonha**
## What does this PR do?

This PR fixes a really annoying bug that would cause the camera to move to the origin when pressing `Alt-Tab` and moving the mouse at the same time (see attached videos for before/after).

## How was this PR tested?

This was tested in the Editor manually. An integration test for the camera was also created to mimic the issue and show that the new code fixes it.

### Before

```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from CameraInputFixture
[ RUN      ] CameraInputFixture.OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate
D:/o3de-2/Code/Framework/AzFramework/Tests/CameraInputTests.cpp(621): error: Value of: m_camera.Translation()
Expected: is close tolerance ((X: 0, Y: -20, Z: 0), 0.01)
  Actual: (X: 0, Y: -10, Z: 0) (of type class AZ::Vector3)
[  FAILED  ] CameraInputFixture.OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate (1 ms)
[----------] 1 test from CameraInputFixture (2 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (3 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] CameraInputFixture.OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate
```

### After

```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from CameraInputFixture
[ RUN      ] CameraInputFixture.OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate
[       OK ] CameraInputFixture.OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate (1 ms)
[----------] 1 test from CameraInputFixture (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (2 ms total)
[  PASSED  ] 1 test.
```

### Before

https://user-images.githubusercontent.com/82228511/190201107-e896ab0f-bfa4-4160-9e2c-11df2cd6ea57.mp4

### After

https://user-images.githubusercontent.com/82228511/190201131-412f30c2-48a3-4cde-96c0-c32d862683cc.mp4

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Viewport/CameraInput.cpp [hunk 1]
```diff
@@ -275,7 +275,8 @@ namespace AzFramework
         {
             auto& cameraInput = m_idleCameraInputs[idleIndex];
             const bool canBegin = cameraInput->Beginning() &&
-                AZStd::all_of(m_activeCameraInputs.cbegin(), m_activeCameraInputs.cend(),
+                AZStd::all_of(m_activeCameraInputs.cbegin(),
+                              m_activeCameraInputs.cend(),
                               [](const auto& input)
                               {
                                   return !input->Exclusive();
```
Comment (hultonha) [inline Code/Framework/AzFramework/AzFramework/Viewport/CameraInput.cpp:282]: This is down to the change in .clang-format for `BinPackArguments: false` - the changes are pretty minor so I've kept them in this PR

File: Code/Framework/AzFramework/AzFramework/Input/Channels/InputChannelDigitalWithSharedModifierKeyStates.h [hunk 2]
```diff
@@ -53,14 +53,13 @@ namespace AzFramework
...
 
-    private:
         ////////////////////////////////////////////////////////////////////////////////////////////
         //! Set the active state of the specified modifier key
         //! \param[in] modifierKey The modifier key to set
         //! \param[in] active The active state to set
         void SetActive(ModifierKeyMask modifierKey, bool active);
-        friend class InputChannelDigitalWithSharedModifierKeyStates;
 
+    private:
```
Comment (hultonha) [inline Code/Framework/AzFramework/AzFramework/Input/Channels/InputChannelDigitalWithSharedModifierKeyStates.h:62]: The previous class was impossible test test so I decided to make this interface public - the chances of misuse are pretty minor and now it's easy to mock `ModifierKeyStates` in a unit/integration test

File: Code/Framework/AzFramework/Tests/CameraInputTests.cpp [hunk 3]
```diff
@@ -575,4 +596,25 @@ namespace UnitTest
...
+    // of the camera if updates are dropped (due to the editor losing focus)
+    TEST_F(CameraInputFixture, OrbitRotateCameraInputDoesNotResetPositionWithInconsitentEventAndUpdate)
+    {
+        const AzFramework::ModifierKeyStates orbitModifierKeystate = OrbitModifierKeyStates(m_orbitChannelId);
+
+        const auto cameraStartingPosition = AZ::Vector3::CreateAxisY(-20.0f);
+        m_targetCamera.m_pivot = cameraStartingPosition;
+        m_pivot = AZ::Vector3::CreateAxisY(-10.0f);
+
+        HandleEvent(AzFramework::InputState{ AzFramework::DiscreteInputEvent{ m_orbitChannelId, AzFramework::InputChannel::State::Began },
```
Comment (hultonha) [inline Code/Framework/AzFramework/Tests/CameraInputTests.cpp:611]: note: This is a best attempt to reproduce the problem where previously multiple calls to HandleEvent (caused by strange ordering or events during Alt-Tab) to happen. The new code is now simpler and more robust and this should hopefully catch the problem if future changes are made.

**Review by hultonha — COMMENTED** [review]
Adding some additional context to PR

**Review by cgalvan — APPROVED** [review]
Nice catch!

**Review by jonawals — APPROVED** [review]
LGTM

**Pull request merged**